dvc/fs/dvc.py

```diff
@@ -333,21 +333,20 @@ def ls(  # pylint: disable=arguments-differ # noqa: C901
                     fs, fs_path, detail=True, ignore_subrepos=ignore_subrepos
                 ):
                     fs_infos[fs.path.name(info["name"])] = info
+                fs_exists = True
             except (FileNotFoundError, NotADirectoryError):
                 pass
 
-            fs_exists = bool(fs_infos) or fs.exists(fs_path)
+        if not (dvc_exists or fs_exists):
+            # broken symlink or TreeError
+            raise FileNotFoundError(errno.ENOENT, os.strerror(errno.ENOENT), path)
 
         dvcfiles = kwargs.get("dvcfiles", False)
 
         infos = []
         paths = []
         names = set(dvc_infos.keys()) | set(fs_infos.keys())
 
-        if not names and (dvc_exists or fs_exists):
-            # broken symlink or TreeError
-            raise FileNotFoundError(errno.ENOENT, os.strerror(errno.ENOENT), path)
```

Comment:
This and the other dropped lines above all come from https://github.com/iterative/dvc/pull/9231. Not sure if I'm missing anything @efiop, but seems like an empty directory was the only way to reach this before and I don't think that was the intent.

Reply:
Thank you for the heads up. I'll take a look and will check tests.

Reply:
This was checking if `ls` returned a non-empty list (which fails for an empty directory) and the path was found in either filesystem. Instead, it seems like it should be checking if the path was NOT found in either filesystem (see the comment above).